src/GafferSceneUI/SceneGadget.cpp

```diff
@@ -656,6 +643,71 @@ size_t SceneGadget::objectsAt(
 	return paths.size();
 }
 
+std::optional<V3f> SceneGadget::normalAt( const IECore::LineSegment3f &lineInGadgetSpace ) const
+{
+	V3f centerP;
+	ScenePlug::ScenePath centerPath;
+
+	auto viewportGadget = ancestor<ViewportGadget>();
+
+	V2f centerRasterP = viewportGadget->gadgetToRasterSpace( lineInGadgetSpace.p0, this );
+	centerRasterP.x = (float)( (int)centerRasterP.x ) + 0.5f;
+	centerRasterP.y = (float)( (int)centerRasterP.y ) + 0.5f;
+
+	if( objectAt( viewportGadget->rasterToGadgetSpace( centerRasterP, this ), centerPath, centerP ) )
+	{
+		// Find the normal of the first triangle formed by `pixelPosition` and
+		// left + top, right + top, left + bottom or right + bottom depth samples
+		// where each pair has the same object ID as the center pixel.
```

Comment:
I think this comment is out of date now, since we no longer only take the first?

Reply:
Yes it is, fixed in https://github.com/GafferHQ/gaffer/pull/5630/commits/4e0abf9ee03fba8599f5047db82b7299a833075d